src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils';
+import { IconButton } from '../Buttons';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/carousel.css';
+
+export type CarouselProps = {
+  currentIndex?: number;
+  autoPlay?: boolean;
+} & React.ComponentPropsWithoutRef<'div'>;
```

Comment:
Think `ref` wont be needed?

Reply:
`ref` never comes from props anyway. I can forwardRef the component, but that still doesn't require changing the props here

Reply:
I forwarded in all subcomponents but left it here because it was messing with the dot notation. Typescript doesn't like if you try attaching subcomponents to a forwardRef component

![image](https://user-images.githubusercontent.com/9084735/158599087-c28867e7-cd30-4a70-8d5a-8f5ea37723ed.png)